src/functionalTest/java/uk/gov/hmcts/reform/pbis/integration/ServiceBusClientTest.java

```diff
@@ -0,0 +1,92 @@
+package uk.gov.hmcts.reform.pbis.integration;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.microsoft.azure.servicebus.IMessage;import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import uk.gov.hmcts.reform.pbis.categories.IntegrationTests;
+
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest
+@Category(IntegrationTests.class)
+public class ServiceBusClientTest extends AbstractServiceBusTest {
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+    }
+
+    @Override
+    @After
+    public void tearDown() throws Exception {
+        super.tearDown();
+    }
+
+    @Test
+    public void receiveMessage_should_receive_next_message_from_subscription() throws Exception {
+        String[] sentMessageContents = {"message1", "message2", "message3"};
+
+        for (String content : sentMessageContents) {
+            serviceBusFeeder.sendMessage(content);
+        }
+
+        List<IMessage> receivedMessages = Stream
+            .generate(() -> receiveMessage())
+            .limit(sentMessageContents.length)
+            .collect(Collectors.toList());
+
+        assertThat(receivedMessages).doesNotContainNull();
+        assertThat(contents(receivedMessages)).containsExactlyInAnyOrder(sentMessageContents);
+    }
+
+    @Test
+    public void completeMessage_should_remove_message_from_the_queue() throws Exception {
+        serviceBusFeeder.sendMessage("content1");
+        serviceBusFeeder.sendMessage("content2");
+
+        IMessage completed = receiveMessage();
+        IMessage notCompleted = receiveMessage();
+
+        assertThat(completed).isNotNull();
+        assertThat(notCompleted).isNotNull();
+        assertThat(receiveMessage()).isNull();
+
+        completeMessage(completed);
+
+        // wait for message locks to expire
+        Thread.sleep(testConfig.getMessageLockTimeoutInMs());
+
+        // verify that the only message in subscription is the not completed one
+        IMessage receivedMessage = receiveMessage();
+        assertThat(receivedMessage).isNotNull();
+        assertThat(content(receivedMessage)).isEqualTo(content(notCompleted));
+        assertThat(receivedMessage.getMessageId()).isEqualTo(notCompleted.getMessageId());
+        assertThat(receiveMessage()).isNull();
```

Comment:
that's a lot of assertions, perhaps soft assertions would add some value? (`.as(...)` may also be useful)

Reply:
Not here - if the first assertion breaks, we don't want other assertions to be evaluated. `.as()` is a good idea - I'll add it to appropriate places.

Reply:
ok